File: formats/gzip/compressor/cs/deflate.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace compressortest
{
	class Deflate
	{
		static void Main(string[] args)
		{
			MemoryStream raw = new MemoryStream();

			Stream stdin = System.Console.OpenStandardInput();
			byte[] chunk = new byte[10240];

			int n = stdin.Read(chunk, 0, chunk.Length);
			while (n > 0)
			{
				raw.Write(chunk, 0, n);
				n = stdin.Read(chunk, 0, chunk.Length);
			}

			byte[] packed = new com.jbase.framework.io.GZipCompressor().compress(raw.ToArray());

			System.Console.OpenStandardOutput().Write(packed, 0, packed.Length);
		}
	}
}

File: formats/gzip/compressor/java/Deflate.java
import java.io.*;
import java.text.*;
import java.util.zip.*;

class Deflate
{
	static public void main(String argv[])
	{
		try 
		{
			ByteArrayOutputStream unpacked = new ByteArrayOutputStream();
			
			// Read data from standard input into memory
			byte[] chunk = new byte[102400];
			int n = System.in.read(chunk, 0, chunk.length);
			while (n > 0)
			{
				unpacked.write(chunk, 0, n);
				n = System.in.read(chunk, 0, chunk.length);
			}
			
			byte[] packed = new com.jbase.framework.io.GZipCompressor().compress(unpacked.toByteArray());

			System.out.write(packed, 0, packed.length);
		} 
		catch (java.io.IOException e)
		{
		}
	}
}
